### PR TITLE
Memory hotplug support

### PR DESCRIPTION
**Pull request by xutian**

### CODE REVIEW

**Comment by xutian** [comment]
ID: 1211916

File: virttest/qemu_devices/qdevices.py [hunk 1]
```diff
@@ -790,3 +790,163 @@ def set_param(self, option, value, option_type=None):
...
+    def verify_hotplug(self, out, monitor):
+        out = monitor.info("memory-devices")
+        memdev = "/object/%s" % self.get_qid()
+        if memdev in out:
+            return True
+        return False
+
+
+class Dimm(QDevice):
+
```
Comment (CongSmile) [inline virttest/qemu_devices/qdevices.py]: Hi @xutian 

It's better to add some description to the class to help understand.

**Comment by xutian** [comment]
[root@dhcp-10-29 virttest]# ../../../tools/scan_results.py 
Test                                                                                                                                                                                              Status     Seconds  Info
----                                                                                                                                                                                              ------     -------  ----
        (Result file: /root/test/autotest-devel/client/results/default/status)
virt.qemu.smp_2.4096m.repeat1.pre_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_native    GOOD       931      completed successfully
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.ping.default_ping                                                     GOOD       903      completed successfully
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.ping.ext_host                                                         GOOD       903      completed successfully

**Comment by lmr** [comment]
Hi @xutian. As you might know, we're moving virt-test to avocado-vt, so I please ask you to consider re-implementing this functionality in the later. The new repo is:

https://github.com/avocado-framework/avocado-vt

Please keep in mind the following guidelines when creating the new PR:

1) Most of the time you can pick the patches of your original PR and merge them on a new branch with minimal changes.
3) Don't use autotest APIs, since we're moving away from using them. Please review the code and change things like `utils.system` -> `process.system`, `utils.run` -> `process.run`, so on and so forth.

Thanks, and I'm sorry for any inconveniences we might have caused your team.

**Pull request closed without merging**